test/helpers.lua

```diff
@@ -816,6 +812,21 @@ function module.read_nvim_log(logfile, ci_rename)
   return log
 end
 
+module.is = {
+  os = {
+    bsd = is_os('bsd'),
+    freebsd = is_os('freebsd'),
+    mac = is_os('mac'),
+    openbsd = is_os('openbsd'),
+    win = is_os('win'),
```

Comment:
I can see that this probably helps with code-completion. But if we go down this road of crafting tables/submodules for code-completion engines, that establishes a forever-pattern just to solve the current limitations of code-completion engines. 

Instead, can code-completion engines discover the available keys using something like emmy lua docstring synstax? E.g. (fictional syntax):

```
-- @param name string: 'bsd'|'mac'|'win'
function is_os(name)
```

Reply:
> Instead, can code-completion engines discover the available keys using something like emmy lua docstring synstax?

Given enough time, probably yeah. I don't mind this pattern regardless, so works for me.

Reply:
Principles/conventions I generally go by:
- less stringy code should be preferred
- function calls should be minimised
- constant information should be calculated up front

Reply:
> Code is always better than comments.

Sure, but these type declarations _are code_ that happens to "fail gracefully" into an inert comment depending on tooling.

> If anything, this is less noisy than what was there before.

I'm not seeing much noise when I look at the `is_os` usages. The noise I'm referring to is the cute `an.english.sentence` pattern that inverts the idea of putting things in meaningful buckets. That pattern used by busted/luassert isn't found in our codebase and is a complete departure and not necessarily desirable for the reasons mentioned in https://github.com/neovim/neovim/pull/21020#discussion_r1020769001